### ISSUE TITLE
Sharing does not work without Source of Truth

### ISSUE THREAD

**Issue by digitalbuddha** [friendlyrobotnyc/Store#47]
It was observed that creating a store with only a fetcher and memory cache stream does not emit when a new get request is made.  Please see sample code below

**Comment by digitalbuddha** [comment]
here is how I discovered issue the following code does not call the streams with each get request


```
class StreamActivity : AppCompatActivity(), CoroutineScope {

    override val coroutineContext: CoroutineContext = Job() + Dispatchers.Main

    @FlowPreview
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.stream_activity)

        var counter = 0

        val store = StoreBuilder
                .fromNonFlow<Int, Int, Int> { key -> key * 1000 + counter++ }
                .cachePolicy(
                        MemoryPolicy
                                .builder()
                                .setExpireAfterWrite(10)
                                .setExpireAfterTimeUnit(TimeUnit.SECONDS)
                                .build()
                )
                .build()

        findViewById<View>(R.id.get_1).onClick {
            store.get(1)
        }

        findViewById<View>(R.id.fresh_1).onClick {
            store.fresh(1)
        }

        findViewById<View>(R.id.get_2).onClick {
            store.get(2)
        }

        findViewById<View>(R.id.fresh_2).onClick {
            store.fresh(2)
        }

        launch {
            store.stream(StoreRequest.cached(1)).collect {
                findViewById<TextView>(R.id.stream_1).text = "Stream 1 $it"
            }
        }
        launch {
            store.stream(StoreRequest.cached(2)).collect {
                findViewById<TextView>(R.id.stream_2).text = "Stream 2 $it"
            }
        }
        launch {
            flowOf(store.stream(StoreRequest.cached(1)),
                    store.stream(StoreRequest.cached(2)))
                    .flattenMerge()
                    .collect {
                        findViewById<TextView>(R.id.stream).text = "Stream $it"
                    }
        }
    }

    fun View.onClick(f: suspend () -> Unit) {
        setOnClickListener {
            launch {
                f()
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        coroutineContext.cancel()
    }
}
```

**Comment by yigit** [comment]
ok so the problem is that the fetcher ends and i've removed the in memory source of truth, which means stream ends :man_facepalming: . Will bring back an implementation, add tests, thanks!

**Comment by digitalbuddha** [comment]
closed by #48
